sync/util/etcd.go

```diff
@@ -14,24 +14,23 @@ import (
 var log = l.NewLogger()
 
 // CreateFromConfig creates etcd sync from config
-func CreateFromConfig(config *util.Config) (s sync.Sync, err error) {
+func CreateFromConfig(config *util.Config) (etcdSync sync.Sync, err error) {
 	syncType := config.GetString("sync", "etcd")
+	etcdServers := config.GetStringList("etcd", nil)
+	if etcdServers == nil {
+		err = fmt.Errorf("no etcd found in config file")
+		return
+	}
```

Comment:
Basically, gohan can work without etcd. These lines force gohan to work with etcd.